### PR TITLE
Added basic extensions.json file

### LINKED ISSUE #94: Simplify contributing and code quality with extensions.json
vscode offers a way to store so called "recommended" extensions.
You can install all recommended extension with one task through a popup,
which is displayed after cloning/opening the repo in vscode.

We could add recommended extension for contributing to vscode,
like tslint which i didn't know about until @kubawolanin tolde me about in a pull request conversation.

Further reading:
https://code.visualstudio.com/docs/editor/extension-gallery#_workspace-recommended-extensions

### PR DESCRIPTION
**Pull request by Confectrician**
Fixes #94 

@kubawolanin can you give me the correct identifier for the tree-view extension?
I can add it to the json file then.

Signed-off-by: Jerome Luckenbach <github@luckenba.ch>

### CODE REVIEW

**Comment by kubawolanin** [comment]
Sure thing, I believe it's `DaGhostman.vs-treeview`.
Marketplace: https://marketplace.visualstudio.com/items?itemName=DaGhostman.vs-treeview
GitHub: https://github.com/DaGhostman/vscode-tree-view

**Comment by Confectrician** [comment]
Then i already got the right one but wasn't sure. 
Tree view is a very common term and gets many results in the extension search.

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "recommendations": [
+        "eg2.tslint",
```
Comment (kubawolanin) [inline .vscode/extensions.json]: Why is TSLint a recommended package for openHAB? There's no *.ts files in OH configuration folder.

Comment (Confectrician) [reply .vscode/extensions.json]: This is shown, when you are working *on the extension itself*.
So "recommended for developing the extension"

Comment (Confectrician) [reply .vscode/extensions.json]: If we would want somethign for openHAB itself we would have to add an `.vscode` folder with that file in the OH configuration folder.

Comment (kubawolanin) [reply .vscode/extensions.json]: Fair enough! Thanks for explanation ;)

**Pull request merged**